### PR TITLE
Created `Secret` type struct

### PR DESCRIPTION
**Pull request by chuckluv**

### CODE REVIEW

File: secret.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package zebra
+
+type Secret struct {
```
Comment (shravyanandyala) [inline secret.go:3]: If this struct only contains a string, we can just make it a type string
```
type Secret string
```

File: secret.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	secret string
+}
+
+func (s *Secret) MarshalText() ([]byte, error) {
+	return []byte("*****"), nil
+}
+
+func (s *Secret) UnmarshalText(text []byte) error {
```
Comment (shravyanandyala) [inline secret.go:11]: If we choose to use ```type Secret string```, we actually don't even need this because json.Unmarshal will figure it out for us.

**Review by shravyanandyala — CHANGES_REQUESTED** [review]

File: secret.go (hunk 1 above)
Comment (rchamarthy) [reply secret.go:3]: right now its just string but we may add some hash algorithm etc to this down the line so ok to be a struct.

File: secret.go (hunk 2 above)
Comment (rchamarthy) [reply secret.go:11]: Yes but struct may be required for potential encryption and hashing. Should be ok for now as its trivial to unmarshal as such.

**Review by rchamarthy — APPROVED** [review]
@chuckluv Lets add a commit that will make the auth.User PasswordHash and Key fields to be of type Secret.

**Review by shravyanandyala — APPROVED** [review]

**Pull request merged**